### DISCUSSION TITLE
Add LLM call duration on session and trace table

### DISCUSSION THREAD

**Discussion by Manouchehri** [langfuse/langfuse#1082] [Ideas]
So I have two generations in this session, which took 18.28 seconds and 6.09 seconds. Which means the LLM calls took 24.37 seconds in total together.

![image](https://github.com/langfuse/langfuse/assets/7232674/331aea5d-bedb-419c-8975-ffd16f583f70)

The session page shows that my session took 24.96 seconds, which is probably technically true (meaning my code took 0.59 seconds to run), but doesn't immediately inform me about how slow (or fast) the LLMs/generations took. For example, if I uploaded a file to a really slow S3 bucket, it would not be obvious if it was the LLM or the S3 bucket that was causing the overall duration to be high.

![image](https://github.com/langfuse/langfuse/assets/7232674/3af6c046-9aa8-4e00-97de-643669abe20e)

To help clarify this, I think an **additional column** labeled "**Generations Duration**" would help highlight the difference between the entire session's runtime vs. generation runtime.

**Comment by marcklingen** [comment]
Thank you for the suggestion. Let's convert this into a feature request instead of a bug :)

I agree completely. The duration of LLM calls is clearly visible on the generation table and within a trace, but it is not visible on the trace and sessions table view.

**Reply by marcklingen** [reply]
@Manouchehri, done. I also slightly edited the body, hope that's ok
